Fix: reset streak for habits after rebirth

Xaz16 opened the pull request:
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes https://github.com/HabitRPG/habitica/issues/11334

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)
- Modify current test of reseting streaks
- Add new if branch for task with "habit" type


[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: d94eca66-c6de-423c-a864-9cb103b9eb77

Xaz16 committed:
Fix: reset streak for habits after rebirth

- Modify current test of reseting streaks
- Add new if branch for task with "habit" type

Alys commented:
@Xaz16 If you test this in your local install, I think you'll find that it doesn't cause the Habit streaks to be reset. I know I said in the issue that the streaks should be reset in the same was as for Dailies, but I'm guessing that was confusing - sorry about that! The code won't be exactly the same for Habits as for Dailies.

Create some Habits in your local install, score them up and down, then examine the data for those Habits in the database, either by doing a `db.tasks.find` mongo search (there'll be some advice about running mongo searches in the wiki) or by using your local install's API to fetch your test user's tasks. By examining the data and comparing it to what you see in your local instal's website, you'll see how streaks on Habits are recorded and you can use that information to work out the code to use to reset the streaks.
You could also look at the data schema for Habits (and the other task types) in the `website/server/models/task.js` file.

Xaz16 committed:
Fix: reset habits by another keys (counterDown/counterUp):

- Fix test for correct way of reseting habits
- Use another keys for reseting habits

Xaz16 commented:
@Alys 
Sure, sorry it is my fault. I don't know why i'm not checked models. I'm tested it with "streak" key reseting of habit and I was really sure that I see how it was reset to zero. 

Now I'm checked model, make fix for habit and checked it locally

Xaz16 committed:
Merge remote-tracking branch 'upstream/develop' into develop

Xaz16 committed:
Merge remote-tracking branch 'upstream/develop' into develop
